addons/mail/tests/test_mail_channel_partner.py

```diff
@@ -222,3 +226,53 @@ def test_channel_public(self):
+    # UNREAD COUNTER TESTS
+    # ------------------------------------------------------------
+
+    def test_unread_counter_with_message_post(self):
+        channel_as_user_1 = self.env['mail.channel'].with_user(self.user_1).create({
+            'name': 'Secret channel',
+            'public': 'public',
+            'channel_type': 'channel',
+        })
+        channel_as_user_1.with_user(self.user_1).add_members(self.user_1.partner_id.ids)
+        channel_as_user_1.with_user(self.user_1).add_members(self.user_2.partner_id.ids)
+        channel_1_rel_user_2 = self.env['mail.channel.partner'].search([
+            ('channel_id', '=', channel_as_user_1.id),
+            ('partner_id', '=', self.user_2.partner_id.id)
+        ])
+        self.assertEqual(channel_1_rel_user_2.message_unread_counter, 0, "should not have unread message initially as notification type is ignored")
+
+        channel_as_user_1.message_post(body='Test', message_type='comment', subtype_xmlid='mail.mt_comment')
+        channel_1_rel_user_2 = self.env['mail.channel.partner'].search([
+            ('channel_id', '=', channel_as_user_1.id),
+            ('partner_id', '=', self.user_2.partner_id.id)
+        ])
+        self.assertEqual(channel_1_rel_user_2.message_unread_counter, 1, "should have one unread message after someone else posted a message")
+
+    def test_unread_counter_with_message_post_multi_channel(self):
+        channel_1_as_user_1 = self.env['mail.channel'].with_user(self.user_1).create({
+            'name': 'wololo channel',
+            'public': 'public',
+            'channel_type': 'channel',
+        })
+        channel_2_as_user_2 = self.env['mail.channel'].with_user(self.user_2).create({
+            'name': 'walala channel',
+            'public': 'public',
+            'channel_type': 'channel',
+        })
+        channel_1_as_user_1.add_members(self.user_2.partner_id.ids)
+        channel_2_as_user_2.add_members(self.user_1.partner_id.ids)
+        channel_2_as_user_2.add_members(self.user_3.partner_id.ids)
+        channel_1_as_user_1.message_post(body='Test', message_type='comment', subtype_xmlid='mail.mt_comment')
+        channel_2_as_user_2.message_post(body='Test', message_type='comment', subtype_xmlid='mail.mt_comment')
+        members = self.env['mail.channel.partner'].search([('channel_id', 'in', (channel_1_as_user_1 + channel_2_as_user_2).ids)], order="id")
+        self.assertEqual(members.mapped('message_unread_counter'), [
+            0,  # channel 1 user 1: posted last message
+            0,  # channel 2 user 2: posted last message
+            1,  # channel 1 user 2: received one message (from message post)
+            2,  # channel 2 user 1: TODO why 2? This is wrong, if notification type is ignored, there is only 1 message in the channel
+            2,  # channel 2 user 3: TODO why 2? This is wrong, if notification type is ignored, there is only 1 message in the channel
```

Comment:
@phenix-factory I've cleaned up the test a bit and added the dependency to the compute

But there 2 lines are wrong. There is only one non-notification message in each channel, it can't be 2.